### ISSUE TITLE
Can k3s please support ingress-nginx on HA clusters? 

### ISSUE THREAD

**Issue by victor-sudakov** [k3s-io/k3s#5629]
**Is your feature request related to a problem? Please describe.**
Yes and no. I suppose ingress-nginx is not officially supported on K3S, and it does not work for me in a HA configuration.

**Describe the solution you'd like**
It would be good to have an up-to-date documentation on how to run ingress-nginx on K3S, especially in HA configurations. Or a clear statement that such a configuration is not supported and never will be.

**Describe alternatives you've considered**
I have studied several articles on the Web explaining how to install ingress-nginx in K3S. Some of them suggest the [baremetal](https://kubernetes.github.io/ingress-nginx/deploy/#bare-metal-clusters) version of the manifest which obviously does not work on HA clusters as it uses hostNetwork. Others suggest the [cloud](https://raw.githubusercontent.com/kubernetes/ingress-nginx/controller-v1.2.0/deploy/static/provider/cloud/deploy.yaml) manifest which would not even install cleanly. Of course I have used `--disable traefik` when installing K3S nodes.

**Additional context**
If anyone has a working setup of a HA K3S cluster with ingress-nginx could you please share your configuration or better yet, document your practice on https://rancher.com/docs/k3s/latest/en/

**K3S version**
v1.23.6. I'm using the PostgreSQL datastore for HA.

**Comment by brandond** [comment]
The upstream helm chart should work fine, which bit of it are you running into trouble with? We're never going to support it in terms of it being a packaged component, but there's no reason it shouldn't work if you deploy the chart properly.

**Comment by victor-sudakov** [comment]
> The upstream helm chart should work fine, which bit of it are you running into trouble with? We're never going to support it in terms of it being a packaged component, but there's no reason it shouldn't work if you deploy the chart properly.

Which exactly chart are you implying? I've tried several, and I'm willing to post the result of running any chart you name.

**Comment by brandond** [comment]
The official one... which one have you been trying?
https://github.com/kubernetes/ingress-nginx

**Comment by brandond** [comment]
It sounds like you might want to configure the chart to deploy as a daemonset, or change the external traffic policy if running it behind a LoadBalancer service?
https://kubernetes.io/docs/concepts/services-networking/service/#traffic-policies

None of these things are k3s related though, they are just decisions about how you want to run and access the ingress.

**Comment by victor-sudakov** [comment]
Nope, `helm install --set controller.kind=DaemonSet ...  ingress-nginx/ingress-nginx` does not fix the timeout on one of the nodes. That's why I was asking for a tested, documented best practice.

**Comment by victor-sudakov** [comment]
> if running it behind a LoadBalancer service?

I actually was not planning to. Maybe just a haproxy choosing one of the K3S nodes which is finds alive.

**Comment by victor-sudakov** [comment]
The way I created my nodes:
```
curl -sfL https://get.k3s.io | sh -s - server --docker --disable traefik \
   --token=XXXXXX --datastore-endpoint="postgres://postgres:XXX@YYY/kubernetes"
```

**Comment by brandond** [comment]
Deploying to k3s shouldn't be different than deploying to any other mutlinode cluster. I believe there are a couple guides out there if you look around. For example: https://www.suse.com/support/kb/doc/?id=000020082

**Comment by victor-sudakov** [comment]
> Deploying to k3s shouldn't be different than deploying to any other mutlinode cluster.

And yet there exist [environment-specific installation instructions](https://kubernetes.github.io/ingress-nginx/deploy/) for many popular Kubernetes implementations. Even separate manifests for different clouds and implementations are provided in the link above.

> I believe there are a couple guides out there if you look around. 

And many of them are outdated or misguided or unsuitable for a multinode K3S cluster. I've checked a number of them, believe me.

> For example: https://www.suse.com/support/kb/doc/?id=000020082

It basically boils down to installing from the standard https://kubernetes.github.io/ingress-nginx repo (as you advised above) with the following non-default values:
```YAML
 controller:
      kind: DaemonSet
      hostNetwork: true
      hostPort:
        enabled: true
      service:
        enabled: false
      publishService:
        enabled: false
      metrics:
        enabled: true
        serviceMonitor:
          enabled: true
      config:
        use-forwarded-headers: "true"
```
I'll check it out today and report here
